wallet/src/de/schildbach/wallet/ui/ToolsActivity.kt

```diff
@@ -48,5 +70,86 @@ class ToolsActivity : BaseMenuActivity() {
             analytics.logEvent(AnalyticsConstants.Settings.NETWORK_MONITORING, bundleOf())
             startActivity(Intent(this, NetworkMonitorActivity::class.java))
         }
+
+        show_xpub.setOnClickListener {
+            handleExtendedPublicKey()
+        }
+
+        var isSyncing = false
+        viewModel.blockchainState.observe(this) {
+            isSyncing = it?.replaying == true
+        }
+
+        transaction_export.setOnClickListener {
+            if (isSyncing) {
+                val dialog = AdaptiveDialog.create(
+                    null,
+                    getString(R.string.report_transaction_history_not_synced_title),
+                    getString(R.string.report_transaction_history_not_synced_message),
+                    getString(R.string.close))
+                dialog.show(supportFragmentManager, "requireSyncing")
+            } else {
+                alertDialog = ExportTransactionHistoryDialogBuilder.createExportTransactionDialog(this,
+                    WalletApplication.getInstance()).buildAlertDialog()
+                alertDialog.show()
+            }
+        }
+    }
+
+    private fun handleExtendedPublicKey() {
+        showExtendedPublicKeyDialog(viewModel.xpubWithCreationDate, viewModel.xpub)
+    }
+
+    private fun showExtendedPublicKeyDialog(xpubWithCreationDate: String, xpub: String) {
+        val view = LayoutInflater.from(this).inflate(R.layout.extended_public_key_dialog, null)
+        val bitmap = BitmapDrawable(
+            resources, Qr.bitmap(xpubWithCreationDate)
+        )
+        bitmap.isFilterBitmap = false
+        val imageView = view.findViewById<ImageView>(R.id.extended_public_key_dialog_image)
+        val xpubView = view.findViewById<TextView>(R.id.extended_public_key_dialog_xpub)
+        imageView.setImageDrawable(bitmap)
+        xpubView.text = xpub
+
+        xpubView.setOnClickListener {
+            handleCopyAddress(xpub)
+        }
+
+        val baseAlertDialogBuilder = BaseAlertDialogBuilder(this)
+        baseAlertDialogBuilder.view = view
+        baseAlertDialogBuilder.negativeText = getString(R.string.button_dismiss)
+        baseAlertDialogBuilder.positiveText = getString(R.string.button_share)
+        baseAlertDialogBuilder.positiveAction = {
+            createAndLaunchShareIntent(xpubWithCreationDate)
+            Unit
+        }
+        alertDialog = baseAlertDialogBuilder.buildAlertDialog()
+        alertDialog.show()
+
+        AdaptiveDialog
```

Comment:
You lost this :) Shouldn't affect anything I guess

Reply:
Thanks for finding this, I was considering how I might use an `AdaptiveDialog` here, but forgot to delete this line.  I will delete it and get a build for QA.